### DISCUSSION TITLE
Simpler way to refresh component

### DISCUSSION THREAD

**Discussion by ryangjchandler** [livewire/livewire#1023] [4. General]
Something I thought would have been included already was a convenient way to refresh and component. 

Currently, you can do this by emitting an event from the component, then registering a listener in the component class that is hooked up to `$refresh`. This seems a bit long winded for just wanting to refresh a component though. 

I've made some changes on a fork that would add a new `$refresh` action: `wire:click="$refresh"`

What are people's thoughts? Happy to make a PR with my changes if it's popular enough?

Changes are here: https://github.com/ryangjchandler/livewire/commit/699b057d36fcd15dbc3d3958f7abe131fa73256b

**Comment by jjjrmy** [comment]
sounds good

**Comment by devcircus** [comment]
I've had to do this on several occasions. Think it's a great idea.

**Comment by baoanhng** [comment]
I didn't know that we couldn't run $refresh directly inside the component itself.

**Reply by ryangjchandler** [reply]
It's a bit of a magic feature.

Inside of your Livewire component you could do:

```php
protected $listeners = ['refreshComponent' => '$refresh'];
```

Then when the `refreshComponent` is emitted, it will refresh the component without running any other actions.

**Reply by baoanhng** [reply]
You missed my point. I thought we could just call `wire:click="$refresh"` inside the component.

**Comment by calebporzio** [comment]
`wire:click="$refresh"` has always worked. I use it all the time!

**Reply by ryangjchandler** [reply]
This is why I closed the PR. I didn't even check that first, rookie error.

**Reply by Wulfheart** [reply]
Man, this is cool. There should be a collection of these little tips and tricks, let's call it: Daily Livewire Snack. 😂

**Reply by bomshteyn** [reply]
How can the `$refresh` be called from the Component class, not from the view?

**Reply by ryangjchandler** [reply]
@bomshteyn - why would you want to call `$refresh` from the class? Any action or method call on the class will automatically trigger a re-render after it has run anyway.

**Reply by bomshteyn** [reply]
@ryangjchandler I want the `mount` method to be re-triggered as well not just `render`. 
Does refresh only trigger render? Is there a reload method that would fully reload the component without full page reload?

**Reply by ryangjchandler** [reply]
Why do you need to re-mount the component? When somebody needs to do that, there's generally a chance to refactor to a method on the class that you can just call.

**Reply by bomshteyn** [reply]
@ryangjchandler  Because i need to have all the previous values to be refreshed from the DB. Yes i could extract all the logic from mount into a seperate function and call it from mount and from the other method. But because its a form and i am tracking the changes to know what changed / needs to be persisted to the DB and other things i would prefer to just reload the component when someone clicks clear changes this way i can be certain that we really are starting from clean slate.

**Reply by jjjrmy** [reply]
@bomshteyn What about refreshing the model?

**Reply by Random72IsTaken** [reply]
> How can the `$refresh` be called from the Component class, not from the view?

Was it using `$wire->call('$refresh');` ?

**Reply by ryangjchandler** [reply]
@GoodM4ven Any time you invoke a Livewire component method / action, it'll refresh the view. There's no need to call it manually.

**Reply by shaffe-fr** [reply]
In my case, data wasn't refreshing because relation on the main model was already set. To fix the issue, I used `unsetRelation` in my actions.

**Comment by SarmadAbdulrahman** [comment]
Good day to all;
Dears i need to know how to work auto-reload component if and update happen in server side ??

**Reply by ju5t** [reply]
You can use [polling](https://laravel-livewire.com/docs/2.x/polling) for that.

Ps. don't hijack other topics. Make your own if you have a completely different question.

**Comment by thenibirahmed** [comment]


**Reply by volkankner** [reply]
this is worked on my situation
